src/components/TimelineComponent.tsx

```diff
@@ -832,11 +856,16 @@ export default class TimelineComponent
   /**
    * Purpose: this function exists explicitly so the core functionality of
    * ttOver cna be tested, as testing d3 events is quite difficult
+   * @precondition: d, x & y are not null
+   * @postcondition: none
    * @param {any} d
    * @param {number} x
    * @param {number} y
    */
   ttOverHelper(d: any, x: number, y: number) {
+    assert(d!=null, 'd should not be null');
+    assert(x!=null, 'x should not be null');
+    assert(y!=null, 'y should not be null');
```

Comment:
I've never seen this before. I think the syntax we adopted was assert.notStrictEqual(x, null, 'x should not be null') or something like that.